contracts/token/ERC721/ERC721.sol

```diff
@@ -386,8 +392,12 @@ abstract contract ERC721 is Context, ERC165, IERC721, IERC721Metadata, IERC721Er
      * Emits an {Approval} event.
      */
     function _approve(address to, uint256 tokenId) internal virtual {
```

Comment:
It is annoying that `approve` is reading the owner twice (the second time in `_approve`). Should we use a similar construction as with `_update` for the operator check?